lib/compiler/passes/generate-js.js

```diff
@@ -684,6 +716,10 @@ function generateJS(ast, options) {
             compileCondition("input.length > peg$currPos", 0);
             break;
 
+          case op.MATCH_ANY_UNICODE:  // MATCH_ANY_UNICODE a, f, ...
+            compileCondition("input.length > peg$currPos", 0);
```

Comment:
This is still the right condition.  If there is one half-surrogate left at the end of the string, `codePointAt()` returns the number for that surrogate.